packages/grafana-data/src/types/logs.ts

```diff
@@ -194,6 +195,35 @@ export enum LogsVolumeType {
   Limited = 'Limited',
 }
 
+/**
+ * Custom meta information required by Logs Volume responses
+ */
+export type LogsVolumeCustomMetaData = {
+  absoluteRange: AbsoluteTimeRange;
+  logsVolumeType: LogsVolumeType;
+  datasourceName: string;
+  sourceQuery: DataQuery;
+};
+
+export const getLogsVolumeAbsoluteRange = (
+  dataFrames: DataFrame[],
+  defaultRange: AbsoluteTimeRange
+): AbsoluteTimeRange => {
+  return dataFrames[0].meta?.custom?.absoluteRange || defaultRange;
+};
+
+export const getLogsVolumeDataSourceInfo = (dataFrames: DataFrame[]): { uid: string; name: string; refId: string } => {
+  return {
+    uid: dataFrames[0]?.meta?.custom?.datasourceUid || '',
+    name: dataFrames[0]?.meta?.custom?.datasourceName || '',
+    refId: dataFrames[0]?.meta?.custom?.sourceQuery?.refId || '',
+  };
+};
+
+export const isLogsVolumeLimited = (dataFrames: DataFrame[]) => {
+  return dataFrames.length && dataFrames[0].meta?.custom?.logsVolumeType === LogsVolumeType.Limited;
```

Comment:
nit:
```suggestion
    return dataFrames[0]?.meta?.custom?.logsVolumeType === LogsVolumeType.Limited;
```

Reply:
Updated in 32e1e5064fb25eb551cf1a1d786df5ffb7fc7211